### DISCUSSION TITLE
VSF2 Shopify calls from behind Google IAP

### DISCUSSION THREAD

**Discussion by jacksinn** [vuestorefront/shopify#220] [Q&A]
Hey there, we have a vsf2 app with a Shopify backend and have issues when behind IAP (we are deploying on app engine) where we then are unable to successfully perform addToCart. Navigating directly to a route or refreshing non-homepage also result in an error (but navigation by clicking around starting from home works fine).

Looking at the console, we can see a [VSF][error] useCart/addItem TypeError: Cannot read properties of null (reading ‘id’). Digging into this we can see on the network tab that the request with the expected payload is sent but the response is an empty object (thus no id property) and a 200 response. Response on local/non-IAP is an object as expected.

Does anyone have any thoughts on fixing these issues behind IAP? Does the vsf2 app integration mask any error and return an empty object? The app works as expected when not behind IAP. I suspect we can set something in nuxt.config but am a bit stuck at this point.

A few days back we were having issue with fetching any data (products, collections) at all from Shopify but after updating the browserBaseURL in the Axios block under publicRuntimeConfig we are able to load that from behind IAP. We fixed that in nuxt.config and expect something similar here.

Thanks for any help.

**Comment by jeffpdotone** [comment]
If you can make a setup to replicate this would be great! :)
